src/vos/vos_common.c

```diff
@@ -638,6 +641,21 @@ vos_metrics_alloc(const char *path, int tgt_id)
 	if (rc)
 		D_WARN("Failed to create 'merged_size' telemetry : "DF_RC"\n", DP_RC(rc));
 
+	/* VOS space SCM used metric */
+	rc = d_tm_add_metric(&vsm->vsm_scm_used, D_TM_GAUGE, "SCM space used", "bytes",
+			     "%s/%s/scm_used/tgt_%u", path, VOS_SPACE_DIR, tgt_id);
+	if (rc)
+		D_WARN("Failed to create 'scm_used' telemetry : "DF_RC"\n", DP_RC(rc));
```

Comment:
so the purpose of these is just to track in our metrics what pmdk (or other libraries) report in their metrics.

I think it would be more useful to me if we also tracked how much space we are using for various parts of the trees.   This is a little tricky because transactions can abort but I did this recently be augmenting allocation logs with a globally set component and then parsing the logs to get a report like this.

value:   88.50 KiB
pool :       0 B
cont :     256 B
obj  :  189.25 KiB
dkey :  312.28 KiB
akey :  321.28 KiB
array:  192.09 KiB
sv   :     512 B
dtx  :  128.00 KiB
total:    1.20 MiB

There are a few complications here though since I needed a hash table to track the category for a given address.  But if we can figure out a way to do something like this, it could be very useful for optimizing our usage.

Reply:
The purpose of this ticket is only to be able to retrieve the space usage per targets.
The goal is to allow end-users to track how the data are spread along the different engines.
@ravalsam , could you confirm this sentence.
From my side, I would prefer to create a new ticket to add these more fine grained metrics.